presidio-image-redactor/presidio_image_redactor/dicom_image_pii_verify_engine.py

```diff
@@ -0,0 +1,336 @@
+
+        Args:
+            all_pos (dict): Dictionary storing all positives.
+            phi_source_dict (dict): Dictionary with PHI labels for
+            this instance.
+            detected_phi (dict): Detected PHI (single entity from
+            analyzer_results).
+            tolerance (int): Tolerance for exact coordinates and size data.
+
+        Return:
+            all_pos (dict): Results dict with all positives, PHI mapped back
+            as possible.
+            match_found (bool): Whether a match was found.
+        """
+        # Get info from detected PHI (positive)
+        results_left = detected_phi["left"]
+        results_top = detected_phi["top"]
+        results_width = detected_phi["width"]
+        results_height = detected_phi["height"]
+        results_score = detected_phi["score"]
+        match_found = False
+
+        # See what in the ground truth this positive matches
+        for label in phi_source_dict:
+            source_left = phi_source_dict[label]["left"]
+            source_top = phi_source_dict[label]["top"]
+            source_width = phi_source_dict[label]["width"]
+            source_height = phi_source_dict[label]["height"]
+
+            match_left = abs(source_left - results_left) <= tolerance
+            match_top = abs(source_top - results_top) <= tolerance
+            match_width = abs(source_width - results_width) <= tolerance
+            match_height = abs(source_height - results_height) <= tolerance
+            matching = [match_left, match_top, match_width, match_height]
+
+            if False not in matching:
+                # If match is found, carry over ground truth info
+                all_pos[label] = phi_source_dict[label]
+                all_pos[label]["score"] = results_score
+                match_found = True
+
+        return all_pos, match_found
+
+    @classmethod
+    def _label_all_positives(
+        cls,
+        gt_labels_dict: dict,
+        ocr_results: dict,
+        analyzer_results: dict,
+        tolerance: int = 50,
+    ) -> dict:
+        """Label all entities detected as PHI by using ground truth and OCR results.
+
+        All positives (analyzer_results) do not contain PHI labels and are thus
+        difficult to work with intuitively. This method maps back to the
+        actual PHI to each detected sensitive entity.
+
+        Args:
+            gt_labels_dict (dict): Dictionary with ground truth labels for a
+            single DICOM instance.
+            ocr_results (dict): All detected text.
+            analyzer_results (dict): Detected PHI.
+            tolerance (int): Tolerance for exact coordinates and size data.
+
+        Return:
+            all_pos (dict): Results dict with all positives, labeled.
+        """
+        all_pos = {}
+
+        # Cycle through each positive (TP or FP)
+        for i in analyzer_results:
+            analyzer_result = analyzer_results[i]
```

Comment:
Doing so just returns the key. For example
```python
tmp = {"1": {"2": [1,2,3], "3": [4,5,6]}}
for analyzer_result in tmp:
    print(analyzer_result)
```
Prints out 1 and not the contents of 1.

Reply:
Disregard. I have switched over to using lists instead of a dictionary so this now works.

Reply:
Right, my bad. In case we could not change the payload, like if it would come from an API, a possible solution would be  using `for analyzer_result in analyzer_results.values():`
Ultimately not using the keys at all would be a 'smell' hinting we can change the structure (as you did)